backend/src/test/java/apsi/team3/backend/controller/integration/EventControllerTest.java

```diff
@@ -0,0 +1,205 @@
+package apsi.team3.backend.controller.integration;
+
+import apsi.team3.backend.DTOs.LoggedUserDTO;
+import apsi.team3.backend.DTOs.Requests.LoginRequest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application-test.properties")
+public class EventControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    static ObjectMapper objectMapper;
+    final static String LOGIN = "testuser1";
+    final static String PASSWORD = "apsi";
+
+    private LoggedUserDTO login() throws Exception {
+        LoginRequest loginRequest = new LoginRequest(LOGIN, PASSWORD);
+        String stringLoginRequest = objectMapper.writeValueAsString(loginRequest);
+        String responseContent = mockMvc.perform(MockMvcRequestBuilders.post("/user/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(stringLoginRequest)
+                )
+                .andReturn().getResponse().getContentAsString();
+        return objectMapper.readValue(responseContent, LoggedUserDTO.class);
+    }
+
+    @BeforeAll
+    public static void setUp() {
+        objectMapper = new ObjectMapper();
+    }
+
+    @Test
+    public void testGetAllEventsUnauthorized() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/events"))
+                .andExpect(MockMvcResultMatchers.status().is4xxClientError())
+                .andExpect(MockMvcResultMatchers.status().isUnauthorized());
+    }
+
+    @Test
+    public void testGetAllEvents() throws Exception {
+        LoggedUserDTO loggedUser = login();
+        String expectedJson = """
```

Comment:
ten test wydaje mi się flaky - ktoś może usunąć te wydarzenia, nie lepiej stworzyć w teście i sprawdzić, czy jest zwrócone? (można poprzez sprawdzenie, czy na liście wszystkich jest to nowe dodane, nie czy cała lista się zgadza)

Reply:
ogólnie się zgadzam, może miałoby to więcej sensu, ale tutaj testy są odpalane na osobnej bazie danych i przed każdym testem jest ona przywracana do stanu początkowego, więc nic złego nie powinno się zadziać

Reply:
makes sense